### PR TITLE
add task

### PR DESCRIPTION
**Pull request by AndriiYelieva**
[DEMO LINK](https://AndriiYelieva.github.io/react_todo-app/)

### CODE REVIEW

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Looks like task not finished.
And please provide demo link in task description

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
.

**Review by etojeDenys — CHANGES_REQUESTED** [review]
1. It looks like this footer does not work
image


2. the editing logic does not implement also

**Comment by AndriiYelieva** [comment]
Help, i have a problem with my footer button

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Still not working, also there are no animation on adding or deleting todos
https://github.com/mate-academy/react_todo-app/assets/61904995/04ee3bfa-fddf-43f0-babe-f670a416d52f

If you have question - ask them in fe_chat

File: src/Components/TodosApp.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { TodosList } from './TodosList';
+import { TodosFooter } from './TodosFooter';
+
+export const TodosApp: React.FC = () => {
+  const { todos } = useContext(TodosContext);
+
+  const isTodos = todos.length > 0;
+
+  return (
+    <TodosProvider>
```
Comment (mykhalenych) [inline src/Components/TodosApp.tsx:15]: why do you have two TodosProvider one in the App?

File: src/hooks/UseLocalStorege.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
+
+export function useLocalStorage<T>(
+  key: string,
+  initialValue: T,
+): [T, (value: T) => void] {
+  const [value, setValue] = useState(() => {
+    const date = localStorage.getItem(key);
+
+    if (date === null) {
```
Comment (mykhalenych) [inline src/hooks/UseLocalStorege.tsx:10]:
```suggestion
    if (!date) {
```

File: src/hooks/UseLocalStorege.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      return initialValue;
+    }
+
+    try {
```
Comment (mykhalenych) [inline src/hooks/UseLocalStorege.tsx:14]: do you have cases when you parse undefined?

**Review by mykhalenych — CHANGES_REQUESTED** [review]
nice work, but I have few questions

**Review by anastasiia-tilikina — CHANGES_REQUESTED** [review]
image


The counter of not completed todos doesn't work properly

File: src/Components/TodosFooter.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        {`${completedTodos} items left`}
+      </span>
+
+      <TodosFilter />
+
+      {completedTodos > 0 && (
+        <button
+          type="button"
+          className="clear-completed"
+          onClick={() => clearCompleted()}
```
Comment (IvanFesenko) [inline src/Components/TodosFooter.tsx:26]:
```suggestion
          onClick={clearCompleted}
```

**Review by IvanFesenko — APPROVED** [review]
Please pay attention on my comments - no need in additional callbacks

**Pull request closed without merging**